File: problem_set/positions-of-large-groups/positions-of-large-groups.java

```
// @Title: 较大分组的位置 (Positions of Large Groups)
// @Author: robert.sunq
// @Date: 2021-08-02 15:44:13
// @Runtime: 1 ms
// @Memory: 38.8 MB

class Solution {
    public List<List<Integer>> largeGroupPositions(String s) {
        List<List<Integer>> res = new ArrayList<>();
        // 数据初始化
        int left = -1,right = -1;
        while(left < s.length() && right < s.length()){
            // 判断指向连续字符的头尾是否包含三个字符
            if(right-left>=2){
                List<Integer> temp = new ArrayList<>();
                temp.add(left);
                temp.add(right);
                res.add(temp);
            }
            // 跳过当前字符串，到达下一位
            left = right+1;
            right = right+1;
            // 将右指针尤以到，下一位不相同的位置
            while(right<s.length()-1 && s.charAt(right) == s.charAt(right+1)){
                right++;
            }
        }
        return res;

    }
}

File: problem_set/positions-of-large-groups/positions-of-large-groups.cpp

```
// @Title: 较大分组的位置 (Positions of Large Groups)
// @Author: robert.sunq
// @Date: 2020-03-20 17:33:08
// @Runtime: 12 ms
// @Memory: 9.2 MB

class Solution {
public:
    vector<vector<int>> largeGroupPositions(string S) {
        vector<vector<int> > an;
        if(S.size() == 0) return an;
        int len = S.size();
        int i=0;
        int j=0;
        while(i<len-2){
            vector <int> indexs;
            int counts =0;
            for(j=i;j<len;j++){
                if(S[i] == S[j]){
                    counts++;
                }
                else break;

            }
            if(counts >= 3){
                indexs.push_back(i);
                indexs.push_back(j-1);
                an.push_back(indexs);
                indexs.clear();
                i = j;
            }
            else i++;
        }

        return an;

    }
};
